runtime/src/main/java/com/flipkart/flux/dao/AbstractDAO.java

```diff
@@ -13,20 +13,39 @@
 
 package com.flipkart.flux.dao;
 
-import com.flipkart.flux.util.HibernateUtil;
 import org.hibernate.Criteria;
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.criterion.Restrictions;
+import javax.inject.Inject;
 
 /**
+ * <code>AbstractDAO</code> class provides methods to perform CRUD operations on an object using Hibernate.
  * @author shyam.akirala
  */
 public class AbstractDAO<T> {
 
+    private SessionFactory sessionFactory;
+
+    @Inject
+    public AbstractDAO(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    /**
+     * Provides the session which is bound to current thread.
+     * @return Session
+     */
     public Session currentSession() {
-        return HibernateUtil.getSessionFactory().getCurrentSession();
+        return sessionFactory.getCurrentSession();
     }
 
+    /**
+     * Retrieves object by it's unique identifier.
+     * @param cls - Class type of the object
+     * @param id
+     * @return (T) Object
+     */
     public T findById(Class cls, String id) {
```

Comment:
In StateMachine and State classes we are having UUID as id, which is of string type. As we would be needing findById in those classes mostly, so kept it in that way. If we need it in future by Long in many places we can add one in AbstractDAO.